File: library/src/main/java/com/getstream/sdk/chat/rest/request/QueryChannelRequest.java
package com.getstream.sdk.chat.rest.request;

import com.google.gson.annotations.SerializedName;

import java.util.Map;

public class QueryChannelRequest {
    @SerializedName("messages")
    Map<String, Object> messages;
    @SerializedName("data")
    Map<String, Object> data;
    @SerializedName("state")
    boolean state;
    @SerializedName("watch")
    boolean watch;

    @SerializedName("subscribe")
    boolean subscribe;

    public QueryChannelRequest(Map messages, Map data, boolean state, boolean watch) {
        this.messages = messages;
        this.data = data;
        this.state = state;
        this.watch = watch;
        this.subscribe = true;
    }
}

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/api2/model/requests/QueryChannelRequest.kt
package io.getstream.chat.android.client.api2.model.requests

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
internal data class QueryChannelRequest(
    val state: Boolean,
    val watch: Boolean,
    val presence: Boolean,
    val messages: Map<String, Any>,
    val watchers: Map<String, Any>,
    val members: Map<String, Any>,
    val data: Map<String, Any>,
)
